src/Evolution/DiscontinuousGalerkin/DgElementArray.hpp

```diff
@@ -14,21 +14,111 @@
 #include "Domain/InitialElementIds.hpp"
 #include "Domain/OptionTags.hpp"
 #include "Domain/Tags.hpp"
+#include "Evolution/Protocols.hpp"
+#include "Evolution/Tags.hpp"
+#include "IO/Importers/VolumeDataReader.hpp"
+#include "IO/Importers/VolumeDataReaderActions.hpp"
 #include "Parallel/ConstGlobalCache.hpp"
 #include "Parallel/Info.hpp"
 #include "Parallel/ParallelComponentHelpers.hpp"
+#include "ParallelAlgorithms/Actions/SetData.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
 #include "Utilities/TMPL.hpp"
 #include "Utilities/TaggedTuple.hpp"
 
+/*!
+ * \brief Don't try to import initial data from a data file
+ *
+ * \see `DgElementArray`
+ */
+struct ImportNoInitialData {};
+
+/*!
+ * \brief Import numeric initial data in the `ImportPhase`
+ *
+ * Requires the `InitialData` conforms to
+ * `evolution::protocols::NumericInitialData`. The `InitialData::import_fields`
+ * will be imported in the `ImportPhase` from the data file that is specified by
+ * the options in the `evolution::OptionTags::NumericInitialData` group.
+ *
+ * \see `DgElementArray`
+ */
+template <typename PhaseType, PhaseType ImportPhase, typename InitialData>
+struct ImportNumericInitialData {
+  static_assert(
+      tt::conforms_to_v<InitialData, evolution::protocols::NumericInitialData>);
+};
+
+namespace DgElementArray_detail {
+
+template <typename DgElementArray, typename InitialData,
+          typename ImportFields = typename InitialData::import_fields>
+using read_element_data_action = importers::ThreadedActions::ReadVolumeData<
+    evolution::OptionTags::NumericInitialData, ImportFields,
+    ::Actions::SetData<ImportFields>, DgElementArray>;
+
+template <typename DgElementArray, typename ImportInitialData>
+struct import_numeric_data_cache_tags {
```

Comment:
what does this struct do? please add or point to the relevant documentation. Is this one just used if not importing numeric initial data? If so, should the second template parameter be `typename ImportNoInitialData`?

Reply:
It's in a detail namespace, so it's an implementation detail. The class template is specialized right after this for `ImportNumericInitialData` where it retrieves the cache tags used by the numeric initial data importer. The non-specialized implementation does nothing. Do you think that's unclear and I should add a comment explaining what happens? I feel like this pattern of template specializations is common enough that we don't need to add a comment about that whenever we do that, don't you think?